src/active-model/model.cr

```diff
@@ -569,6 +583,26 @@ abstract class ActiveModel::Model
     {% end %}
   end
 
+  # Declare an `ActiveModel::Model` attribute
+  #
+  # # General Arguments
+  #
+  # - `persistence`: Will not emit the key when serializing if `false`.
+  # - `converter`: A module defining alternative serialisation behaviour for the attribute's value.
+  #
+  # # Serialisation Target Arguments
+  #
+  # Additionally, `attribute` accepts the following optional arguments for different serialisation targets.
+  #
+  # ## JSON
+  # - `json_key`: Override the emitted JSON key.
+  # - `json_emit_null`: Emit `null` if value is `nil`, key is omitted if `json_emit_null` is `false`.
+  # - `json_root`: Assume value is inside a JSON object under the provided key.
+  # ## YAML
+  # - `yaml_key`: Override the emitted YAML key.
+  # - `yaml_emit_null`: Emit `null` if value is `nil`, key is omitted if `yaml_emit_null` is `false`.
+  # - `yaml_root`: Assume value is inside a YAMLk object under the provided key.
```

Comment:
As `converter` applies to both JSON and YAML serialisation, it would be good to either:
1. provide the ability to specify seperate JSON / YAML converters, or
2. provide the ability to specify `key`, `emit_null` and `root` that's used in both JSON and YAML (unless otherwise specified).

Reply:
I would prefer these options to default to the unprefixed argument, with the option to override via a prefixed argument

Reply:
sorry I'm not exactly sure what this means, could you please clarify?

Reply:
Kim: `json_converter` and `yaml_converter`
Me: Support unprefixed arguments, like `root`, that is passed to both the json and yaml prefixed arguments. If, in my example, `json_root` is defined, use that instead of `root`